lib/data_grid.dart

```diff
@@ -629,3 +642,47 @@ enum SortingState {
     }
   }
 }
+
+class SortingIconSettings {
+  final IconData ascending;
+  final IconData descending;
+  final double size;
+  final Color? color;
+  final EdgeInsets padding;
+
+  const SortingIconSettings({
+    required this.ascending,
+    required this.descending,
+    required this.size,
+    this.color,
+    required this.padding,
+  });
+
+  const SortingIconSettings.defaults()
+      : ascending = Icons.arrow_upward_rounded,
+        descending = Icons.arrow_downward_rounded,
+        size = 20,
+        color = null,
+        padding = const EdgeInsets.symmetric(horizontal: 4.0);
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+
+    return other is SortingIconSettings &&
+        other.ascending == ascending &&
+        other.descending == descending &&
+        other.size == size &&
+        other.color == color &&
+        other.padding == padding;
+  }
+
+  @override
+  int get hashCode {
+    return ascending.hashCode ^
```

Comment:
https://api.dart.dev/stable/2.18.4/dart-core/Object/hash.html

Object.hash(...) provides a better hash I believe. I wonder if we really need to override these operators. I don't see them being compared anywhere

Reply:
It is to reduce build calls. In the case of `SortingIconSettings` not changing between builds, the hash code would be a different memory address and thus an unneeded rebuild.

Reply:
Huh never thought of that. Good point.